docs/DeveloperGuide.md

```diff
@@ -4,26 +4,167 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Application Launch
 
+The following diagram shows a sequence diagram of the program when it is run.
+
+![ApplicationLaunchSequenceDiagram](img/ApplicationLaunchSequenceDiagram.png)
+
+1. `InvMgr` does a setup by creating the required `Ui`, `Storage`, and `ItemList` objects.
+2. To create the `ItemList` object, `load()` from `Storage` must be called. The resulting `ArrayList<Item>` from `load()` is used to create the `ItemList` object.
+3. Then, `InvMgr` will continuously loop, doing the following:
+   i. `InvMgr` will call upon `Ui` to get input from the user.
+   ii. `InvMgr` passes the user input to `Parser.parse(command)`.
+   iii. `Parser.parse(command)` returns a `Command` object.
+   iv. `InvMgr` calls upon the `execute()` method of the returned `Command` object
+4. The loop stops when the user types `exit`.
+
+### UI Component
+![UiClassDiagram](img/UiClassDiagram.png)
+
+The `UI` component consists of a single `Ui` class.
+
+The `UI` component
+- Displays salutations, prompts for user input, error messages and results of queries.
+- Reads in user inputs
+- Depends on the `Messages` and `InvMgrException` classes in the `Common` component. It displays messages stored in the `Messages` class and displays an error message whenever `InvMgrException` is invoked.
+
+### Parser Component
+
+![ParserClassDiagram](img/ParserClassDiagram.png)
+
```

Comment:
![Screenshot 2022-03-31 at 11 27 54](https://user-images.githubusercontent.com/60136330/160970511-5ef013f4-87ba-4791-be7f-ff7be16fe6ee.png)
Consider stating the return types of the methods called in InvMgr?